modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -801,6 +929,12 @@ private void dropTableLocally(long causalityToken, String name, UUID tblId, List
         }
     }
 
+    private List<ClusterNode> calculateAssignments(TableConfiguration tableCfg, int partNum) {
+        Integer partitions = tableCfg.partitions().value();
+        Integer replicas = tableCfg.replicas().value();
+        return AffinityUtils.calculateAssignments(baselineMgr.nodes(), partitions, replicas).get(partNum);
```

Comment:
Oof, this smells so badly. If we only need assignments for a single table, why do we calculate them for potentially hundreds or thousands irrelevant partitions?

Reply:
Right, I'll rewrite it